app/services/irs_attempts_api/tracker.rb

```diff
@@ -1,20 +1,32 @@
 module IrsAttemptsApi
   class Tracker
-    attr_reader :session_id, :enabled_for_session
+    attr_reader :session_id, :enabled_for_session, :request, :user, :sp
 
-    def initialize(session_id:, enabled_for_session:)
-      @session_id = session_id
+    def initialize(session_id:, request:, user:, sp:, enabled_for_session:)
+      @session_id = session_id # IRS session ID
+      @request = request
+      @user = user
+      @sp = sp
       @enabled_for_session = enabled_for_session
     end
 
     def track_event(event_type, metadata = {})
       return unless enabled?
 
+      event_metadata = {
+        user_agent: request&.headers['User-Agent'],
+        unique_session_id: hashed_session_id,
+        user_uuid: AgencyIdentityLinker.for(user: user, service_provider: sp),
+        device_uuid: "FIXME", #user&.device&.cookie_uuid,
```

Comment:
@zachmargolis It's not obvious what the right way to get this data is. I briefly started something like `user.devices.last.cookie_uuid` but that felt like it was very likely wrong. Is this accessible through the session in a way I didn't find?

Reply:
In a controller context it should be `cookies[:device]`, so maybe we need to pass that in from the controller as an argument

Reply:
These two comments were super-helpful; thanks!